src/components/search/search.service.ts

```diff
@@ -69,23 +75,36 @@ export class SearchService {
           matchedProps,
         }))
         // Map Org results to Org & Partner results based on types asked for
-        .flatMap((result) =>
-          result.type !== 'Organization'
-            ? result
-            : [
-                ...(inputTypes.includes('Organization') ? [result] : []),
-                // If matched Organization, include Partner implicitly
-                ...(inputTypes.includes('Partner')
-                  ? [
-                      {
-                        id: result.id, // hydrator knows this is an org id not partner
-                        type: 'PartnerByOrg' as const,
-                        matchedProps: ['organization' as const],
-                      },
-                    ]
-                  : []),
-              ],
-        )
+        .flatMap((result) => {
+          // Map to be used for search results that don't follow standard pattern, add to this as needed
+          const inputTypeMap = {
+            EthnologueLanguage: inputTypes.includes('Language')
+              ? [
+                  {
+                    id: result.id,
+                    type: 'LanguageByEth' as const,
+                    matchedProps: ['ethnologue'] as const,
+                  },
+                ]
+              : [],
```

Comment:
You are missing the case here where EthnologueLanguage is asked for, but it is never returned.
That's what the new line 91 does for Orgs.

Also missing adding the search type implicitly. Like the new line 57 does for Orgs.